apps/site/assets/ts/shuttles/components/ShuttlesPage.tsx

```diff
@@ -1,21 +1,115 @@
-import React, { ReactElement } from "react";
-import ShuttlesOverview from "./ShuttlesOverview";
+import React, { ReactElement, useState, useEffect } from "react";
+import ShuttlesMap from "./ShuttlesMap";
+import StationDropdown from "./StationDropdown";
 import { Route } from "../../__v3api";
+import { Coordinates, Diversion, MaybeDirectionId } from "./__shuttles";
 
 interface Props {
   route: Route;
+  diversions: Diversion;
 }
 
-const ShuttlesPage = ({ route }: Props): ReactElement<HTMLElement> => {
+const ShuttlesPage = ({
+  route,
+  diversions
+}: Props): ReactElement<HTMLElement> => {
   const places =
     route.id === "Green" ? route.direction_names : route.direction_destinations;
 
+  const [selectedDirectionId, setSelectedDirectionId] = useState<
+    MaybeDirectionId
+  >(null);
+  const [displayedShapes, setDisplayedShapes] = useState([
+    ...diversions.shapes
+  ]);
+  const [displayedStops, setDisplayedStops] = useState([...diversions.stops]);
+  const [mapCenter, setMapCenter] = useState<Coordinates>();
+
+  useEffect(
+    () => {
+      if (selectedDirectionId === null) {
+        setDisplayedShapes([...diversions.shapes]);
+        setDisplayedStops([...diversions.stops]);
+      } else {
+        setDisplayedShapes(
+          diversions.shapes.filter(
+            shape => shape.direction_id === selectedDirectionId
+          )
+        );
+        setDisplayedStops(
+          diversions.stops.filter(
+            stop =>
+              stop.direction_id === selectedDirectionId ||
+              stop.direction_id === null
+          )
+        );
+      }
+    },
+    [diversions, selectedDirectionId]
+  );
+
+  const DirectionButton = (
+    directionId: MaybeDirectionId,
+    headsign: string
+  ): JSX.Element => (
+    <button
+      key={`${headsign}`}
+      type="button"
+      className={`btn btn-secondary btn-sm c-btn-group-stackable__btn ${selectedDirectionId ===
+        directionId && "c-btn-group-stackable__btn--active"}`}
+      onClick={() => setSelectedDirectionId(directionId)}
+    >
+      {headsign}
+    </button>
+  );
+
+  const centerMapOnStation = (stationId: string): void => {
```

Comment:
We're inconsistent in some of this code between calling them `stations` or `stops`. I'd say we should stick with `stop` since that is the name of the domain model.

Reply:
I'm going to table this for my next PR because yes.